WebPhapp/backend/server.js

```diff
@@ -215,54 +109,159 @@ app.post('/api/v1/prescriptions/add',(req,res) => {
+            var drug = answer.rows.filter((row) => {
+                return (row.ID === toSend[i].drugID);
+            });
+
+            // Could be undefined on return
+            if(drug.length !== 0)
+              toSend[i].drugName = drug[0].NAME;
         }
 
-        console.log("/api/v1/prescriptions/single: Sent prescription with ID " + prescriptionID);
-        res.json(prescription);
+        console.log(msg);
+        res.json(toSend);
     })
     .catch((error) => {
-        console.log("/api/v1/prescriptions/single: error: ", error);
+        console.log("/api/v1/prescriptions: error: ", error);
         res.json({});
     });
 });
 
+/*
+An api endpoint that returns a single prescription given a prescription ID
+Examples:
+    Directly in terminal:
+        >>> curl "http://localhost:5000/api/v1/prescriptions/single/1"
+    To be used in Axois call:
+        .get("api/v1/prescriptions/single/1")
+Returns:
+    A prescription object each with fields: [
+        prescriptionID,
+        patientID,
+        drugID,
+        fillDates,
+        writtenDate,
+        quantity,
+        daysFor,
+        refillsLeft,
+        prescriberID,
+        dispenserID,
+        cancelDate,
+        drugName
+    ]
+Warning:
+    no validation exists for blockchain index yet. See Issue #32 on GitHub.
+*/
+app.get('/api/v1/prescriptions/single/:prescriptionID', (req,res) => {
+    var prescriptionID = parseInt(req.params.prescriptionID);
+    var handlePrescriptionCallback = function(prescription) {
+        // '==' catches both null and undefined
+        if (prescription == null) {
+            console.log('/api/v1/prescriptions/single: No ID match');
+            res.status(400).send({});
+            return;
+        }
+
+        // if no connection string (Travis testing), fill drugName with dummy info
+        if (!conn.MySQL) {
+            prescription.drugName = "drugName";
+            console.log("/api/v1/prescriptions/single: Sent prescription with ID " + prescriptionID);
+            res.json(prescription);
+            return;
+        }
+
+        // Look up the drug name given the ID in MySQL
+        mysql.getDrugNamesFromIDs([prescription.drugID], connection) // prescription.drugID
+        .then((answer) => {
+            if (answer.rows.length === 0) {
+                prescription.drugName = 'drugName not found in Pharmacopoeia';
+            } else {
+                prescription.drugName = answer.rows[0].NAME;
+            }
+
+            console.log("/api/v1/prescriptions/single: Sent prescription with ID " + prescriptionID);
+            res.json(prescription);
+        })
+        .catch((error) => {
+            console.log("/api/v1/prescriptions/single: error: ", error);
+            res.json({});
+        });
+    };
+
+    if(conn.Blockchain){
+        block_helper.read(prescriptionID)
+        .then((answer) => {
+            handlePrescriptionCallback(answer.prescription);
+        }).catch((error) => {
```

Comment:
This catches call errors. Is it necessarily the case an error leads to a prescription not found?